Applications/Spire/Source/OrderImbalanceIndicator/FilteredOrderImbalanceIndicatorModel.cpp

```diff
@@ -0,0 +1,116 @@
+#include "Spire/OrderImbalanceIndicator/FilteredOrderImbalanceIndicatorModel.hpp"
+
+using Filter = Spire::FilteredOrderImbalanceIndicatorModel::Filter;
+using namespace Nexus;
+using namespace Spire;
+
+FilteredOrderImbalanceIndicatorModel::FilteredOrderImbalanceIndicatorModel(
+  std::shared_ptr<OrderImbalanceIndicatorModel> source_model,
+  std::shared_ptr<std::vector<Filter>> filters)
```

Comment:
```filters``` can not be a ```shared_ptr```.

Reply:
So how does the model store them?

Reply:
Use a ```shared_ptr``` to store them, but not as a parameter.

Reply:
How would that work? Wouldn't that mean creating a shared_ptr from either a stack-allocated variable or a raw pointer?

Reply:
Yep that's right.

Reply:
Doesn't this now create the opposite problem, where the lifetime of the whatever is holding the filters would have to be longer than the filter model? Is the assumption that the filter models would be used in a limited scope?

Reply:
I think the filter model should just take and store the shared_ptr. If the filter model outlives whatever has a handle to the model, then the filters will still be valid in this case.

Reply:
FilteredOrderImbalanceIndicatorModel::FilteredOrderImbalanceIndicatorModel(
      std::shared_ptr<OrderImbalanceIndicatorModel> source_model,
      std::vector<Filter> filters)
      : m_filters(std::make_shared<std::vector<Filter>>(std::move(filters)))